### PR TITLE
Oh God The Zipper's Stuck: Entombed quirk granting unremovable roundstart MODsuit

### PR DESCRIPTION
**Pull request by Ephemeralis**
## About The Pull Request

A neat little twist on the idea of roundstart MODsuits, this PR adds a new quirk called Entombed which traps your character inside a relatively low-grade creation *permanently*. Here's the full details:

- The 'fused' entombed MODsuit has 10 complexity and class III armor to all damage types except wounding, which remains at I.
   - It it is also indestructible and can't be burned or melted off you with acid.
   - This may seem high, but the only way you can upgrade this throughout the course of the round is via modules, and you don't have much complexity to throw around.
- You cannot remove this MODsuit. Ever. The unit can't be stripped from you, either.
- The MODsuit is *very* slow when unpowered, almost twice as slow as an ordinary MOD. Think of this as a soft version of the Hemophage's dormant tumor.
- The appearance (skin), name, description and deployed piece prefix of the suit are all customizable via the quirk preferences menu.
   - Changing the skin does not alter the stats of the MODsuit in any way - it's purely cosmetic.
- If you're an Ethereal, the MODsuit begins play with an ethereal MOD core.
- If you're a Plasmaman, the MODsuit begins play with a unremovable 0 complexity plasma stabilizer module, allowing you to take off ONLY YOUR HELMET and not combust to death.
- All races start with a unremovable 0 complexity modified joint torsion module, which provides a small amount of charge to your suit when you move around. I've balanced this around providing just enough to make the *base* suit not really lose much charge while you're actively moving. That goes right out of the window once you start adding modules that draw passive charge of your own, though.
- (NOT ADDED YET) If your MODsuit's charge runs out, you take a hefty lump of toxin damage every tick until you charge it up again, to reflect your physiology or the suit's life support systems shutting down.

There are no forced limits on deploying parts - I initially thought about locking all but the head parts into permanent deployment, but I think that'd kind of suck for some concepts and so have opted to leave it to the player to decide.

## How This Contributes To The Nova Sector Roleplay Experience

This opens up a slew of concepts, from infirm characters relying on specialized life support systems via exosuits, to slimes who can't maintain humanoid form without the rigidity of a suit, to engineer plasmamen who've colonized an entire modsuit with a skeleton stuck inside it and become inexorably linked. Maybe you play a veteran who has deep-rooted trauma preventing them from leaving the safety of their suit.

It's really up to you. Importantly, it's *cool* and customizable.

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>
  
IZ3JqWFbPm

dreamseeker_K7iBskzCaC

dreamseeker_I2gOX0synx

</details>

## Changelog

:cl: yooriss
add: The Entombed quirk has been added, allowing characters to start off with a permanently unremovable low-end MODsuit stuck to their back slot. Letting the suit's charge run low will eventually kill you, and the quirk has special interactions with both Ethereals and Plasmamen!
/:cl:

### CODE REVIEW

**Comment by SomeRandomOwl** [comment]
The only thing i can think of is...what if the person dies and medical needs to do surgery on them, if it cannot be removed then it will be impossible to revive them, as like a soft lock DNR. I don't know if that's intentional or if there is anyway around that?

**Comment by Ephemeralis** [comment]
> The only thing i can think of is...what if the person dies and medical needs to do surgery on them, if it cannot be removed then it will be impossible to revive them, as like a soft lock DNR. I don't know if that's intentional or if there is anyway around that?

There's medical MOD modules that will help with this and you can still strip to retract the pieces, it just won't remove the actual MOD unit itself.

**Comment by Nerev4r** [comment]
God I wish we had a prison-themed MODsuit for this. This is so cool.

**Comment by SomeRandomOwl** [comment]
> God I wish we had a prison-themed MODsuit for this. This is so cool.

Na just add the tychus from starcraft 2 as a suit design and this quirk

**Comment by OmegaDarkPotato** [comment]
In the event of gibbing (particularly large explosion, shuttlecrush), what happens if they're brought back in a new body without the MOD?

**Comment by Ephemeralis** [comment]
> In the event of gibbing (particularly large explosion, shuttlecrush), what happens if they're brought back in a new body without the MOD?

The MODsuit has the indestructible flag, so I don't think it can actually be destroyed by any of these things. If it somehow can, they'll take regular toxin damage (once I add that) for the rest of the round. Hooking someone up to an ammoniated mercury IV drip until they return to Central, anyone?

**Comment by W4rd3nn** [comment]
reminds me of warcaskets

**Comment by Ephemeralis** [comment]
> > > In the event of gibbing (particularly large explosion, shuttlecrush), what happens if they're brought back in a new body without the MOD?
> > 
> > 
> > The MODsuit has the indestructible flag, so I don't think it can actually be destroyed by any of these things.
> 
> shuttles will still gib, as shuttles gib everything unless there is some sort of flag to prevent the mob from being gibbed, i think megafauna have this and, for an explosion, the modsuit would need to have enough bomb protection (like the bombsuit) to prevent gibbing/delimbing (or some flag that clothing has that prevents it i dunno)

I tested this and indeed, shuttlegibbing destroys it. My solution is thus: don't get gibbed. If you come back without the MODsuit, be prepared to be on heavy antitox meds for the rest of the shift since that thing was supporting your existence.

**Comment by Lafrien** [comment]
Wait, so you **can** turn off the MOD? Can you just turn it off and live a slow life without being afraid of dying from charge?

Can you change battery on Bluespace?

If it has inbuilt storage, but you can't take it off, how others are supposed to **search** you? Can you just carry Nuclear Disk and illegal items without fear of being searched?

Will you be able to splash it with Slime potion to negate the reduced running speed?

**Comment by Nerev4r** [comment]
I believe if you have any kind of nodrop back storage (this is true for snail shells at least) stripping you just dumps it all on the ground.

**Comment by Namwty** [comment]
Stabilized red and slime speed potions shouldn't work with this, imo, as arguably slowness is biggest downside. 
Best guess is kinda how adamantine armor does it, it ignores slime speed and stabilized red, thats if we care about them being able to speed up.

Also, if settler quirk will bypass the slowness cuz it interacts with item slowdowns, maybe making them incompatible is a good idea? Unless it was already adressed, then cool.


Being gibbed etc and revived in a humonkey+SAD, would just most likely make you lack all quirks and effects they have. Niche cases anyhow, so not a big deal, I imagine.

**Comment by Ephemeralis** [comment]
The slowness is only when the MOD is off - it's a slightly faster than normal MOD when the suit is powered. If you want to slime all your MOD outerwear to give you more mobility if/when your suit power runs out (to get to a charger, for example), I think that's fine.

Settler slowdown is *additive* I think, so it'll just make the slowdown worse if anything. I'll test it, though.

File: modular_nova/master_files/code/modules/entombed_quirk/code/entombed.dm [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+	var/life_support_timer
+	/// Are we taking damage?
+	var/life_support_failed = FALSE
+
+/datum/quirk/equipping/entombed/process(seconds_per_tick)
+	var/mob/living/carbon/human/human_holder = quirk_holder
+	if (!modsuit || life_support_failed)
+		// we've got no modsuit or life support. take damage ow
+		if (isslimeperson(human_holder))
+			human_holder.adjustToxLoss(-ENTOMBED_TICK_DAMAGE * seconds_per_tick, updating_health = TRUE)
```
Comment (Hatterhat) [inline modular_nova/master_files/code/modules/entombed_quirk/code/entombed.dm:30]: this could probably just be using the forced = TRUE flag for toxloss? if that's still a thing. saves you the trouble of having to check for slime status

Comment (Ephemeralis) [reply modular_nova/master_files/code/modules/entombed_quirk/code/entombed.dm]: good catch! added this

**Comment by Namwty** [comment]
> Settler slowdown is _additive_ I think, so it'll just make the slowdown worse if anything. I'll test it, though

Settler has a effect of significantly reducing item slowdowns ( big slappy, beartraps ,bolas, fire suits, bio suits and so on ). For example snails had ( still slighlty has ) issue with settler cause it would bypass the slowness applied by backpack.
Feel like it might be case here, if slowness is tied to modsuit itself, tho not tooo sure.

**Comment by Lafrien** [comment]
> wait if it's toxin damage wouldn't slime people be healed by their suit being off

No, it was already discussed - they will get build-up of toxins, not healed. MOD will check if person is a slimeperson or not.

**Comment by Ephemeralis** [comment]
I've fixed the stripping issue: they'll now mirror the same behavior as snail shells where they drop their contents onto the person's tile when stripped.

With this, I think we're finally good to go!

**Comment by Namwty** [comment]
I will say.
There is concern of someone with this quirk getting put in prison, and security not being able to remove their modules due to unremovable mod. Some modules would make them arguably uncontainable without a headache, like teleporter mod.
Doubt it's too big of a issue, but dunno, depends on players and popularity of quirk.

**Comment by Nerev4r** [comment]
> I will say. There is concern of someone with this quirk getting put in prison, and security not being able to remove their modules due to unremovable mod. Some modules would make them arguably uncontainable without a headache, like teleporter mod. Doubt it's too big of a issue, but dunno, depends on players and popularity of quirk.

There's an implant that stops teleports. It's not an issue.

File: modular_nova/master_files/code/modules/entombed_quirk/code/entombed.dm [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+	life_support_failed = TRUE
+
+/datum/quirk/equipping/entombed/add_unique(client/client_source)
+	. = ..()
+	var/mob/living/carbon/human/human_holder = quirk_holder
+	if (istype(human_holder.back, /obj/item/mod/control/pre_equipped/entombed))
+		modsuit = human_holder.back // link this up to the quirk for easy access
+
+	if (isnull(modsuit))
+		return
```
[the remark this answers is not in the document]
Comment (Ephemeralis) [reply modular_nova/master_files/code/modules/entombed_quirk/code/entombed.dm:67]: Done this! It was decently easy to extend the equipping quirk datum to record what it dropped which should also make this easier for future quirks to do similar things.

File: modular_nova/master_files/code/modules/entombed_quirk/code/entombed.dm [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+		part.name = "[modsuit.theme.name] [initial(part.name)]"
+		part.desc = "[initial(part.desc)] [modsuit.theme.desc]"
+
+	// quickly deploy it on roundstart
+	modsuit.quick_activation()
+	// lower the helmet for style points and also to make chargen less annoying
+	var/obj/item/clothing/head/mod/helmet = locate() in modsuit.mod_parts
+	modsuit.retract(human_holder, helmet)
+
+	// deploy specific racial features - ethereals get ethereal cores, plasmamen get free plasma stabilizer module
```
[the remark this answers is not in the document]
Comment (Ephemeralis) [reply modular_nova/master_files/code/modules/entombed_quirk/code/entombed.dm:117]: Done this!

**Comment by Ephemeralis** [comment]
I committed a minor git cardinal sin with a series of bad rebases. Things should be fine now. I think.

**Comment by radarseas** [comment]
_few things from poking around ingame_

okay, SO... if you somehow lose a limb, the parts of the suit that don't retract fall to the floor, there's one issue though.

This can be done via MANY means..
1: heretic rifts
2: amputation sheers
3: cybernetic limb mounts
4: dying while cursed (burn a bible)
5: mediborg's amputation adventure...
6: pretty much a hundred other means...

These parts of the suit DO NOT DROP... (but they uh, somehow drop on the floor to picked up and the suit can deploy with those parts of the suit missing (uh oh). WHILE FULLY deployed (UH OH).

Suit should always be retractable because urm, yeah. that happens.

The prefs load for the next person that spawns in, from testing we've done in the ghost cafe. This is bad.

**Comment by Ephemeralis** [comment]
I'll look into this, but maybe the perma-deployed thing is a pipe dream because of the above.

**Comment by FarsightedNightlight** [comment]
I love the concept of this quirk! I do have one question though, will there be more options added with time? I noticed the addition of the Infiltrator modsuit, which _is_ antagonist-themed, it simply raises the question if something like the Contractor modsuit would be on the table. Mostly food for consideration, and all that fun.

**Comment by GlaciaImca** [comment]
Or the science modsuit if there is already the antag themed suits in there, why not the research themed one?

**Comment by Ephemeralis** [comment]
I don't really want to make any command-specific MODsuits available for what I hope are obvious reasons - and unfortunately, the Research one counts as that.

We may be able to do contractor as long as it doesn't step on people's toes.

**Comment by GlaciaImca** [comment]
The research one is command specific? I guess that makes sense as to why I have seen it on the wiki but never in play...

Does make me think that science could really use there own mod plating though for non-RD mod users.

**Comment by FarsightedNightlight** [comment]
Another small one, the old mining modsuit skin, which is currently just behind a modsuit paintkit on the mining mod, could also be added! It's a pretty good option to also add.

**Comment by FieryMorty** [comment]
So I found out that in a round, if you name your suit something, like say I named mine "N2 filled", another random player in the round with Entombed gets that name instead. So my friend got the MOD name "N2 filled" instead of me, and I got his name, "Enviro-A.A.D" for his MODsuit instead for some reason.

**Comment by Ephemeralis** [comment]
> So I found out that in a round, if you name your suit something, like say I named mine "N2 filled", another random player in the round with Entombed gets that name instead. So my friend got the MOD name "N2 filled" instead of me, and I got his name, "Enviro-A.A.D" for his MODsuit instead for some reason.

This should *never ever* happen. I'm not actually sure how it can even happen.

**Comment by Ephemeralis** [comment]
I've (re)added a thing to make sure customizations are applied properly. I have a feeling I know why the name thing was happening.

**Comment by Ephemeralis** [comment]
The only way I think I can salvage the full-deploy option is to give the player the no_dismember trait at roundstart with the suit, but that's honestly a fairly decent buff on its own.

File: modular_nova/master_files/code/modules/entombed_quirk/code/entombed.dm [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+
+/datum/preference/choiced/entombed_skin/init_possible_values()
+	return list(
+		"Standard",
+		"Civilian",
+		"Colonist",
+		"Advanced",
+		"Atmospheric",
+		"Corpsman",
+		"Cosmohonk",
```
Comment (OrionTheFox) [inline modular_nova/master_files/code/modules/entombed_quirk/code/entombed.dm:154]: Not sure about cosmohonk as a skin... unless you could like, clown-restrict it.
Yes, I know modules aren't applied, but just the skin itself is kind of... too silly.

File: modular_nova/master_files/code/modules/entombed_quirk/code/entombed.dm [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+		"Engineering",
+		"Infiltrator",
```
Comment (OrionTheFox) [inline modular_nova/master_files/code/modules/entombed_quirk/code/entombed.dm:156]: Just to double-check, the Infiltrator one is proven working with the quirk 100% as intended?
The Infiltrator modsuit is a Belt item with no backpack so it might not behave the same.

**Comment by Namwty** [comment]
So fused mode has some issues

It seems that thread ripper sometimes refuses to function ( happened twice, in my case, when two different medical players tried using them ) while alone it's whatever, and can be shrugged of as "soft DNR", it does cause bothersome issues when amputation shears come into play.
But I do know thread ripper works sometimes, as I've seen it work, but something causes it to not function right... maybe sometype of module, like plating, but that's just me reaching at this point.

Issue becomes clearer when medical tries to use amputation shears and what not, mainly biggest comes from fact that when you lose a limb, it drops modsuit part ( which isn't meant to drop ), which means that if a doctor picks it up accidentally, it glues to their arm and requires admin intervention to remove. 

Also if someone dies from toxins cause of suit vital failure, it seems there is really no way to revive them, cause defib ceases to function due to high toxin dmg and no chemical can purge it in first place. So in short, you need to revive someone to remove toxins, but they have too much toxins to revive in first place. ( Heck even heart of darkness won't even try to revive you in darkness with that damage ). But again, can be shrugged of as "Soft DNR" and a downside of a quirk, which is fine as well, but decided to bring it up anyhow.


TL;DR
Losing limb due to shears or what not, creates item which once picked up... cannot be dropped by any normal means. Even if Fused option is fun and great RP, due to it's issues, might not be ideal to have it as option, to avoid other players from accidentally catching a stray bug ( unless can be fixed ).

**Comment by tiramisuapimancer** [comment]
This might be a big ask, but would it be possible for the fully deployed version of the selected suit appearance to show up on the character preview?

**Comment by Ephemeralis** [comment]
I've made the following changes:

- Colonist has been removed from the list of skin options. Dunno why, but it's just not working and even copying the skin element across doesn't seem to be doing the trick.
- Maximum MODsuit name length increased to 48 chars.

And now for the big one...

- Force-deployed Entombed users gain the NO_DISMEMBER trait.

NO_DISMEMBER is pretty huge as a dismemberment damage is a big part of of what makes some weapons so lethal (like the energy katana, for example. It does however, fix all of the issues involved with force-deployment in regards to things falling off properly and it plays properly with wounds and the like.

I couldn't think of a less severe way to do this, sorry.

**Comment by SovietSpaghetti** [comment]
If you can get an implant then take it out with this quirk it's stuck to your hand. Which is a problem, because you can't just remove your limb to get it off.

**Comment by Guidesu** [comment]
A issue is that they delete the loadouts items. And if you use the option to put it all in the case it does not applies correctly the color changes and name/desc changes. So what the modsuit essentially does is delete the suit, shoes and gloves

**Comment by Ephemeralis** [comment]
> A issue is that they delete the loadouts items. And if you use the option to put it all in the case it does not applies correctly the color changes and name/desc changes. So what the modsuit essentially does is delete the suit, shoes and gloves

I've fixed this. An annoying foible of the loadout system, but what can one do.

**Comment by Ephemeralis** [comment]
> "If you can get an implant then take it out with this quirk it's stuck to your hand. Which is a problem, because you can't just remove your limb to get it off." This still an issue?

I'm not actually sure what this is referencing. Nothing is adding TRAIT_NODROP to implants, or messing with any overrides to produce this behavior. I'm not even sure how an implant would get TRAIT_NODROP in the first place...

**Comment by Ephemeralis** [comment]
I think so, yeah. Haven't had any major reports of weirdness after the no_dismember addition.

**Comment by bigbro175** [comment]
you can use it as a prisoner, seems like that shouldn't be a thing. perhaps limit it from that role specifically, looks problematic.
